src/JSInterop/Microsoft.JSInterop/src/Implementation/JSDataReference.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.JSInterop.Implementation
+{
+    /// <summary>
+    /// Implements functionality for <see cref="IJSDataReference"/>.
+    /// </summary>
+    public sealed class JSDataReference : JSObjectReference, IJSDataReference
```

Comment:
nit: How do you feel about changing the name to JSBufferReference or similar that is more explicit about this being a buffer rather than "Data"

Reply:
If we were renaming this, I'd propose `JSStreamReference`.

I think it's an improvement over "buffer" because it better aligns with the consumption experience. You receive it as a stream, not a contiguous block of memory (like most buffers). And unlike most scenarios where we say "buffer", you can't mutate it in some way that's visible back on the other side. You can't mutate the stream at all.

The reason I originally proposed `JSDataReference` was an attempt to be more neutral. On the sending side, it's a buffer, and on the receiving side, it's a stream. The only word I could think of that covered both of them was simply "data". However since the code here is on the .NET side where you are getting a stream, I'd be OK with `JSStreamReference`. It's a bit confusing because you may think you don't have a JS stream to send.

Overall I think I'd be OK with either `JSStreamReference` or `JSDataReference`.

Reply:
Fair enough, "Data" feels "too generic" to me, but its fine. 

Thinking more about this, we should expose an overload that exposes the pipe reader directly to avoid unnecessary memory copies.

Reply:
903b16dc5a